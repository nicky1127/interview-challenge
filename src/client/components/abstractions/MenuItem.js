import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { removeItem } from "../../redux/_actions";

const MenuItem = props => {
  const { item, idx } = props;
  let content;
  if (Array.isArray(item.dietaries) && item.dietaries.length > 0) {
    content = item.dietaries.map((dietary, index) => (
      <span
        key={`menuItem_dietary-${index}`}
        className="dietary"
        data-testid="menuItem_dietary"
      >
        {dietary}
      </span>
    ));
  }

  const onClickBtn = () => {
    props.removeItem(idx);
  };
  return (
    <li className="item" data-testid="menuItem">
      <h2>{item.name}</h2>
      <p>{content}</p>
      <button className="remove-item" onClick={onClickBtn}>
        x
      </button>
    </li>
  );
};

MenuItem.propTypes = {
  removeItem: PropTypes.func
};

const ConnectedMenuItem = connect(null, { removeItem })(MenuItem);

export default ConnectedMenuItem;
