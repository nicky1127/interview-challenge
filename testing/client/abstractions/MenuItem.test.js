import React from "react";
import { render, cleanup } from "@testing-library/react";
import MenuItem from "client/components/abstractions/MenuItem";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

beforeEach(() => {
  jest.clearAllMocks();
  cleanup();
});

const props = { item: { name: "test", id: 1, dietaries: ["test"] }, idx: 0 };

const state = {
  selectedItemIDs: [],
  items: [{ name: "test", id: 1, dietaries: ["test"] }],
  ldgItems: false,
  ldgSuccItems: false
};
const mockStore = configureMockStore();
const store = mockStore(state);

describe("MenuItem", () => {
  it("Snapshot matchs", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MenuItem {...props} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("MenuItem should render one [data-testid=MenuItem]", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <MenuItem {...props} />
      </Provider>
    );
    const container = getByTestId("menuItem");
    expect(container).toBeInTheDocument();
  });
});
