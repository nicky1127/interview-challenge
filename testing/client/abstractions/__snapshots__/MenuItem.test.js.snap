// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem Snapshot matchs 1`] = `
<li
  className="item"
  data-testid="menuItem"
>
  <h2>
    test
  </h2>
  <p>
    <span
      className="dietary"
      data-testid="menuItem_dietary"
    >
      test
    </span>
  </p>
  <button
    className="remove-item"
    onClick={[Function]}
  >
    x
  </button>
</li>
`;
